lib/utilities/serverAccessLogger.js

```diff
@@ -0,0 +1,395 @@
+        'objectDeleteTagging': 'OBJECT',
+        'objectGet': 'OBJECT',
+        'objectGetACL': 'OBJECT',
+        'objectGetLegalHold': 'OBJECT',
+        'objectGetRetention': 'OBJECT',
+        'objectGetTagging': 'OBJECT',
+        'objectCopy': 'OBJECT',
+        'objectHead': 'OBJECT',
+        'objectPut': 'OBJECT',
+        'objectPutACL': 'OBJECT',
+        'objectPutLegalHold': 'OBJECT',
+        'objectPutTagging': 'OBJECT',
+        'objectPutPart': 'OBJECT',
+        'objectPutCopyPart': 'OBJECT',
+        'objectPutRetention': 'OBJECT',
+        'objectRestore': 'OBJECT',
+        // 'serviceGet': '',
+        // 'websiteGet': '',
+        // 'websiteHead': '',
+    });
+
+    return `REST.${req.method}.${methodToResType[req.apiMethod] ? methodToResType[req.apiMethod] : 'UNKNOWN'}`;
+}
+
+function getRequester(authInfo) {
+    const requester = null;
+    if (authInfo) {
+        if (authInfo.isRequesterPublicUser && authInfo.isRequesterPublicUser()) {
+            return requester; // Unauthenticated requests
+        } else if (authInfo.isRequesterAnIAMUser && authInfo.isRequesterAnIAMUser()) {
+            // IAM user: include IAM user name and account
+            const iamUserName = authInfo.getIAMdisplayName ? authInfo.getIAMdisplayName() : '';
+            const accountName = authInfo.getAccountDisplayName ? authInfo.getAccountDisplayName() : '';
+            return iamUserName && accountName ? `${iamUserName}:${accountName}` : authInfo.getCanonicalID();
+        } else if (authInfo.getCanonicalID) {
+            // Regular user: canonical user ID
+            return authInfo.getCanonicalID();
+        }
+    }
+    return requester;
+}
+
+function getURI(request) {
+    let requestURI = null;
+    if (request) {
+        const method = request.method || 'UNKNOWN';
+        const url = request.url || request.originalUrl || '/';
+        const httpVersion = request.httpVersion || '1.1';
+        requestURI = `${method} ${url} HTTP/${httpVersion}`;
+    }
+    return requestURI;
+}
+
+function getObjectSize(request, response) {
+    const objectSizePutMethods = Object.freeze({
+        'objectPut': true,
+        'objectPutPart': true,
+    });
+
+    const objectSizeGetMethods = Object.freeze({
+        'objectGet': true,
+    });
+
+    // If it is a PUT get the Content-Length from the request, if it is a GET get it from the response.
+    if (request && response && objectSizeGetMethods[request.apiMethod]) {
+        const len = response.getHeader('Content-Length');
+        return len || null;
+    }
+
+    if (request && objectSizePutMethods[request.apiMethod]) {
+        const len = request.headers['content-length'];
+        return len || null;
+    }
+
+    return null;
+}
+
+function getBytesSent(res, bytesSent) {
+    if (bytesSent) {
+        return bytesSent;
+    }
+
+    if (!res) {
+        return null;
+    }
+
+    const len = res.getHeader('Content-Length');
+    return len || null;
```

Comment:
what if Content-Length is 0, should we log it as 0 and not null?
A test should have caught this potential bug.